code/game/objects/items/storage/backpack.dm

```diff
@@ -610,7 +609,19 @@
 		to_chat(wearer, "<span class='danger'>Your cloak shimmers from the damage!</span>")
 		apply_shimmer()
 
-/obj/item/storage/backpack/marine/satchel/scout_cloak/proc/action_taken(datum/source, atom/target, obj/item/I, mob/living/wearer)
+/obj/item/storage/backpack/marine/satchel/scout_cloak/proc/on_gun_firing(datum/source, atom/target, obj/item/weapon/gun/firing_gun)
+	action_taken()
+
+/obj/item/storage/backpack/marine/satchel/scout_cloak/proc/on_gun_attachment_firing(datum/source, atom/target, obj/item/attachable/firing_attachment, obj/item/weapon/gun/master_gun)
```

Comment:
these are pointless, if you dont care about the arguments just use one proc like it was before

Reply:
Clarity. The people seeing the proc need to know correctly what parameters they are actually receiving, even if they are not currently being used.

Reply:
I disagree. someone else can merge this as I won't.

Reply:
It's up to the individual if they want to make a different proc for each signal if the alternative is a receiving proc that takes 0 (or 1 for the source) arguments